src/Data/Functor/Foldable.hs

```diff
@@ -197,19 +198,45 @@ class Functor (Base t) => Recursive t where
   -- | A variant of 'cata' in which recursive positions also include the
   -- original sub-tree, in addition to the result of folding that sub-tree.
   --
-  -- A very simple example is collecting all recursive children of a value
+  -- This is useful for updating a deeply-nested node and then cascading the
+  -- change on the path from that node to the root: when processing a node
+  -- along that path, use the updated version of the next node along that path,
+  -- and the untouched version of the other child nodes.
   --
-  -- >>> let tree = Node "a" [Node "b" [Node "c" [], Node "d" []], Node "e" [], Node "f" []]
-  -- >>> showTree tree
-  -- (a (b c d) e f)
+  -- >>> :{
+  -- >>> deleteFirstLeaf :: Tree a -> Tree a
+  -- >>> deleteFirstLeaf = para $ \case
+  -- >>>   NodeF a (x@(Node _ [], _) : xs)  -- x is a leaf
+  -- >>>     -> Node a (fmap fst xs)  -- drop it, leave xs as-is (fst)
+  -- >>>   NodeF a (x : xs)  -- x is not a leaf
+  -- >>>     -> Node a (snd x : fmap fst xs)  -- recur (snd) on x, leave xs as-is (fst)
+  -- >>>   NodeF a []  -- this is a leaf
+  -- >>>     -> Node a []  -- leave it as-is
+  -- >>> :}
   --
-  -- >>> traverse_ showTree $ para (\fun -> embed (fmap fst fun) : foldMap snd fun) tree
-  -- (a (b c d) e f)
-  -- (b c d)
-  -- c
-  -- d
-  -- e
-  -- f
+  -- >>> let tree = Node "a" [Node "b" [Node "X" [], Node "c" []], Node "d" [], Node "e" []]
+  -- >>> putStr $ drawTree tree
+  -- a
+  -- |
+  -- +- b
+  -- |  |
+  -- |  +- X
+  -- |  |
+  -- |  `- c
+  -- |
+  -- +- d
+  -- |
+  -- `- e
+  -- >>> putStr $ drawTree $ deleteFirstLeaf tree
+  -- a
+  -- |
+  -- +- b
+  -- |  |
+  -- |  `- c
+  -- |
+  -- +- d
+  -- |
+  -- `- e
```

Comment:
I didn't use `showTree` because in `(a (b X c) d e)`, it looks like the left-most child is `a`, not `X`. I am guessing you'll say that this is taking up too much space?

Reply:
> I am guessing you'll say that this is taking up too much space?

Yes. That was an idea of using custom `showTree`.

I don't see how this example is "better".

In fact in your example, I'd write a function of type `TreeF a (Tree a, b) -> TreeF a b` as a transformation.
I.e. first define `paraHoist n = para (embed . n)`, and use that.

With children example, we don't unfold into resulting list. it could be a `Set` for example, or any other `Monoid`. Maybe changing it to set (and having duplicate nodes) would make that example even nicer.

Reply:
> I don't see how this example is "better".

As I've explained in [the other issue](https://github.com/ekmett/recursion-schemes/pull/74#issuecomment-640281285) and in this PR's description, I don't want to release as-is because the documentation for `para` and the README's mention of `para` contradict each other. One of the two ways of fixing the inconsistency is to change the `para` example to match the README; that's what this PR does.

> I'd [...] first define `paraHoist n = para (embed . n)`, and use that.

Ok, so that confirms that's it's the README which is incorrect, not the `para` example, as it is `paraHoist`, not `para`, which names the common recursion pattern of path copying.

How do you suggest to change the README? I see a couple of possibilities:

1. add `paraHoist` (under the name `pathCopying` or `copyPath`?) to `recursion-schemes`, and change the README to refer to it instead of `para`.
2. Change the example. I thought that was a really good example, but it is now clear to me that it is not one which recursion-schemes users recognize in practice. Do you have a better example from your own experience?
3. Drop the "Clarity" benefit from the list? That would be a shame, because that's the most convincing benefit for me.

Reply:
Re `paraHoist` name. it's terrible one I came up quickly with. I don't know literature at all, maybe it has a name. (It's probably some instantiation of `ghylo`). I don't think we should rush to add it to the library.

I personally wasn't unhappy with Edwards usual (very brief) README (i.e. what we had). So from those options I'd pick 3rd as the easiest way forward.

---

I had some time to spend on `recursion-schemes` during the weekend, but now I once don't. I'm very sorry, but now I'll prefer "whatever get's the release out of the door".

If it means reverting my documentation commit, I'll be fine with that. Yet, don't let perfect kill good.

Reply:
> now I'll prefer "whatever get's the release out of the door".
> 
> If it means reverting my documentation commit, I'll be fine with that

I think a good compromise would be to keep most of the documentation improvements, but to remove the `para` documentation until we have a better story for it.

Reply:
fixed in 7e70c2641fa00f80105d151281957ea3a32e2e1b